addons/mail/models/discuss/discuss_channel.py

```diff
@@ -659,11 +661,7 @@ def message_post(self, *, message_type='notification', **kwargs):
         if (not self.env.user or self.env.user._is_public()) and self.is_member:
             # sudo: discuss.channel - guests don't have access for creating mail.message
             self = self.sudo()
```

Comment:
I think the main issue with the other sudo is that this forgot to take into account portal user, the fix should be here.